File: ProtoSASOnScala/src/main/scala/dd/sas/pipeline/calculation/structures/flows/generativeflows/ObjectFlow.scala
package dd.sas.pipeline.calculation.structures.flows.generativeflows

import dd.sas.pipeline.calculation.structures.flows.Flow
import dd.sas.pipeline.calculation.structures.nodes.Node
import dd.sas.pipeline.worldmodel.WorldObject

/**
  * Created by Sergey on 25.09.2016.
  */
abstract class ObjectFlow[O <: WorldObject](node: Node) extends Flow(node){
  def apply(): List[O]
}

File: ProtoSAS/src/main/java/dd/sas/pipeline/calculation/structures/flows/generativeflows/ObjectFlow.java
package dd.sas.pipeline.calculation.structures.flows.generativeflows;

import dd.sas.pipeline.calculation.structures.flows.Flow;

import java.util.List;
import java.util.function.Consumer;

/**
 * Created by Sergey on 13.09.2016.
 */
public interface ObjectFlow<T> extends Flow {
    List<T> getResult();
    void subscribe(Consumer<ObjectFlow<T>> consumer);
}
